server/infinispan/src/test/java/org/hawkular/apm/server/infinispan/InfinispanDynamicPropertyCacheTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.server.infinispan;
+
+import java.util.Arrays;
+import java.util.UUID;
+
+import org.hawkular.apm.server.api.services.CacheException;
+import org.hawkular.apm.server.api.services.ObservableProperty;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author Pavol Loffay
+ */
+public class InfinispanDynamicPropertyCacheTest extends AbstractInfinispanTest {
+
+
+    @Test
+    public void testGetNonExisting() {
+        InfinispanDynamicPropertyCache cache = createCache();
+
+        ObservableProperty observableProperty = cache.get(null, "id");
+        Assert.assertNull(observableProperty);
+    }
+
+    @Test
+    public void testGetOne() throws CacheException {
+        InfinispanDynamicPropertyCache cache = createCache();
+
+        ObservableProperty<Double> doubleObservableProperty = new ObservableProperty<>("id", 15D);
+        ObservableProperty<String> stringObservableProperty = new ObservableProperty<>("id2", "str");
+
+        cache.store(null, Arrays.asList(doubleObservableProperty, stringObservableProperty));
+
+        ObservableProperty propertyFromDb = cache.get(null, "id");
+        Assert.assertEquals(doubleObservableProperty.getValue(), propertyFromDb.getValue());
+
+        propertyFromDb = cache.get(null, "id2");
+        Assert.assertEquals(stringObservableProperty.getValue(), propertyFromDb.getValue());
+    }
+
+    @Test
+    public void testSystemProp() {
+        InfinispanDynamicPropertyCache cache = createCache();
+
+        System.setProperty("strProp", "foo");
```

Comment:
Prefix the property with the test name, or unset it after the test :) Otherwise, another test, somewhere, might use "strProp" as well and wonder why it's failing. Worse yet: this might fail only on CI.